### ISSUE TITLE
range select on columns

### ISSUE THREAD

**Issue by ericlangedijk** [JAM-Software/Virtual-TreeView#928]
It would be very nice if the treeview - in gridmode - would support a multi-selection of columns.

**Comment by chuacw** [comment]
Hmm, I suppose my commit would close this issue too?

**Comment by joachimmarder** [comment]
> I suppose my commit would close this issue too?

How exactly would one do this? I tried this in your new demo project `Multiselect`. Of course I can manually select all cells of multiple columns, but I guess the idea was a bit different. And the reporter wanted to bind this ability to the option `toGridExtensions` being set.

**Comment by chuacw** [comment]
We can select multiple columns by clicking on the first row in the first column then shift clicking on the last row in the last column.

**Comment by joachimmarder** [comment]
> We can select multiple columns by clicking on the first row in the first column then shift clicking on the last row in the last column.

For me that isn't column selecting but just cell selecting. How would you select the 2nd and 4th column?

**Comment by chuacw** [comment]
In most applications, it's just click on the 2nd column header, then ctrl+click on 4th column header.

**Comment by joachimmarder** [comment]
Yes, that is common in most applications, but I didn't see this in your sample. So I was wondering if this isue here is truly solved as intended.

**Comment by chuacw** [comment]
@joachimmarder I see.

No, I didn't implement column selection in that sense then.

Column selection is ambiguous. 
One has to think whether when you click on the header, are you asking for column sort, or are you asking for column selection?
If it's shift click, it's clear one is requesting for adding the column to the existing selection.

**Comment by joachimmarder** [comment]
> Column selection is ambiguous.

Yes, that is indeed a problem.

> If it's shift click, it's clear one is requesting for adding the column to the existing selection.

Yes, that sounds good. Maybe Ctrl would be even better because Shift is typically used for a range-selection.

But this was not yet implemented in the PR, right @chuacw?

**Comment by chuacw** [comment]
"But this was not yet implemented in the PR, right @chuacw?" 

@joachimmarder It's nearly there 😅 Probably need another 20-30 lines of code to get there...
